### PR TITLE
Fix redoing after append transaction

### PR DESCRIPTION
**Pull request by RichieAHB**
This is a proposed solution to https://github.com/ProseMirror/prosemirror/issues/819, it also requires https://github.com/ProseMirror/prosemirror-state/pull/13 for it to work (although won't break if this patch isn't present i.e. a user is using a `prosemirror-state` without prior to that hypothetical version)

N.B. the new test here naturally won't pass without the change to `prosemirror-state`.

### CODE REVIEW

File: src/history.js [hunk 1]
```diff
@@ -260,16 +260,17 @@ function applyTransaction(history, state, tr, options) {
 
   if (tr.getMeta(closeHistoryKey)) history = new HistoryState(history.done, history.undone, null, 0)
 
-  let appended = tr.getMeta("appendedTransaction")
+  let rootTr = tr.getMeta("appendedTransaction")
   if (tr.steps.length == 0) {
     return history
-  } else if ((appended || tr).getMeta("addToHistory") !== false) {
+  } else if (tr.getMeta("addToHistory") !== false) {
```
Comment (marijnh) [inline src/history.js:266]: I'd like to keep the behavior where transactions appended to a non-history root transaction aren't added to history either. What is the reason for this change?

**Comment by marijnh** [comment]
I do see the problem here, but I'm not sure yet what the right solution is. I suspect this patch makes it possible for the history to become corrupted (you preserve the old `undone` stack, but that is no longer aligned with the current document, since the transaction changed that). I'll think about it more when I'm back from my holiday.

**Comment by RichieAHB** [comment]
Yeah I realised this but thought I’d leave the straw man up as I think it
makes the problem a bit clearer than I put it in words. A plugin could do
anything to the state, I guess the lazy assumption by me here is that it
probably shouldn’t, but it actually isn’t a sound assumption. Enjoy your
holiday!

File: src/history.js (hunk 1 above)
Comment (RichieAHB) [reply src/history.js:266]: As far as I can see this is the same thing. A transaction's `appendedTransaction` [can only be itself](https://github.com/ProseMirror/prosemirror-state/blob/545244357dbc12cc1f340cd3f8ad75dc7472a803/src/state.js#L131) (if at all). So `appended` is either falsey [or `=== tr`](https://github.com/RichieAHB/prosemirror-history/blob/22f773640ce79c91f328d14ef20de8361b2fb7f0/src/history.js#L263), unless there's somewhere else that `appendedTransaction` is being set that I've missed?

Comment (marijnh) [reply src/history.js:266]: > A transaction's appendedTransaction can only be itself (if at all).

Right, but that's a bug (the behavior as you fixed it in the other PR was what I was _trying_ to implement there)

**Comment by RichieAHB** [comment]
Ah right - I did wonder! Either way it can wait 👍

**Comment by marijnh** [comment]
Closing in favor of 2a94750
